lib/PersonalityProvider.jsm

```diff
@@ -15,6 +15,42 @@ const {RecipeExecutor} = ChromeUtils.import("resource://activity-stream/lib/Reci
 ChromeUtils.defineModuleGetter(this, "NewTabUtils",
   "resource://gre/modules/NewTabUtils.jsm");
 
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+ChromeUtils.import("resource://gre/modules/XPCOMUtils.jsm");
+ChromeUtils.defineModuleGetter(this, "OS", "resource://gre/modules/osfile.jsm");
+XPCOMUtils.defineLazyGlobalGetters(this, ["fetch"]);
+
+XPCOMUtils.defineLazyGetter(this, "gTextDecoder", () => new TextDecoder());
+
+// This doesn't work currently, the fetch wasn't returning the expected
+// data structure from https://kinto.dev.mozaws.net/v1/, failing on
+// capabilities not being defined.
+
+XPCOMUtils.defineLazyGetter(this, "baseAttachmentsURL", async () => {
+  const server = Services.prefs.getCharPref("services.settings.server");
+  const serverInfo = await (await fetch(`${server}/`)).json();
+  const {capabilities: {attachments: {base_url}}} = serverInfo;
+  return base_url;
+});
+
+const PERSONALITY_PROVIDER_DIR_NAME = "personality-provider";
```

Comment:
you could do this I think? 
`const PERSONALITY_PROVIDER_DIR = OS.Path.join(OS.Constants.Path.localProfileDir, "personality-provider")`

Reply:
Good catch.